File: app/src/main/java/com/moufee/purduemenus/di/AppComponent.java
package com.moufee.purduemenus.di;

import android.app.Application;

import com.moufee.purduemenus.MenusApp;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * Defines where dependency injection may be performed
 */
@Singleton
@Component(modules = {AndroidSupportInjectionModule.class, AppModule.class, ActivityModule.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }

    void inject(MenusApp app);

}

File: app/src/main/java/com/moufee/purduemenus/di/AppComponent.kt
package com.moufee.purduemenus.di

import android.app.Application
import com.moufee.purduemenus.MenusApp
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Defines where dependency injection may be performed
 */
@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, AppModule::class, ActivityModule::class, WorkerModule::class])
interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder
        fun build(): AppComponent
    }

    fun inject(app: MenusApp)
}